src/dmd/expressionsem.d

```diff
@@ -3387,7 +3388,7 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
             return setError();
         }
 
-        size_t nargs = exp.arguments ? exp.arguments.dim : 0;
+        const hasArgs = exp.arguments ? exp.arguments.dim > 0 : false;
```

Comment:
This refactoring appears to have nothing to do with the PR. It also winds up computing that expression 3 times instead of once.

Reply:
The unclear error messages were a result of calling preFunctionParameters on exp.arguments, which called expandTuples.  Moving PFP to later means args aren't expanded yet, so computing nargs here could give us the wrong answer if there are tuples.